File: .claude/hooks/ruff-format.mjs
#!/usr/bin/env node
// Formate le fichier Python avec ruff après écriture (PostToolUse Write|Edit)
import { readFileSync } from 'fs';
import { execSync } from 'child_process';
import { fileURLToPath } from 'url';

function defaultExec(cmd) {
  execSync(cmd, { stdio: 'ignore', timeout: 15_000 });
}

export function run(input, { exec = defaultExec } = {}) {
  const filePath = input.tool_input?.file_path ?? input.tool_input?.path ?? '';
  if (!filePath.endsWith('.py')) return null;

  try {
    exec(`uv run ruff format "${filePath}"`);
    return null;
  } catch {
    // uv/ruff absent — non bloquant
    return null;
  }
}

/* v8 ignore next 4 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  const input = JSON.parse(readFileSync(0, 'utf8'));
  run(input);
}

File: .claude/hooks/ruff-format.py
#!/usr/bin/env python3
# @hookstack post-write-ruff-format
"""Formats the Python file with ruff after write (PostToolUse Write|Edit)."""
import json
import subprocess
import sys


def _exec(cmd, timeout=15):
    return subprocess.run(
        cmd, shell=True, capture_output=True, text=True, timeout=timeout, check=True
    )


def run(input_data, exec_cmd=None):
    if exec_cmd is None:
        exec_cmd = _exec
    tool_input = input_data.get("tool_input") or {}
    file_path = tool_input.get("file_path") or tool_input.get("path") or ""
    if not file_path.endswith(".py"):
        return None

    try:
        exec_cmd(f'uv run ruff format "{file_path}"')
    except Exception:
        # uv/ruff absent — non bloquant
        pass
    return None


if __name__ == "__main__":
    data = json.load(sys.stdin)
    run(data)
